omnipaxos/src/ballot_leader_election.rs

```diff
@@ -208,80 +204,78 @@ impl BallotLeaderElection {
     pub(crate) fn hb_timeout(
         &mut self,
         seq_paxos_state: &(Role, Phase),
-    ) -> (Ballot, ElectionStatus) {
+        seq_paxos_promise: Ballot,
+    ) -> (ElectionStatus, Ballot) {
+        if seq_paxos_promise > self.leader {
+            // Sync leader with Paxos promise in case ballot didn't make it to BLE followers
+            self.leader = seq_paxos_promise;
+        }
         self.update_leader();
         self.update_happiness(seq_paxos_state);
-        self.update_leader_state(seq_paxos_state);
         self.update_ballot();
+        let election_status = self.get_election_status(seq_paxos_state);
         self.new_hb_round();
-        (self.leader, self.election_status)
+        (election_status, self.leader)
     }
 
     fn update_leader(&mut self) {
         self.leader = self
             .heartbeat_replies
             .iter()
             .fold(self.leader, |a, b| a.max(b.ballot));
```

Comment:
This fold looks a bit weird, can't we just do `max_by`? 
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by